packages/web-config-server/src/apiV1/DataAggregatingRouteHandler.js

```diff
@@ -59,12 +50,36 @@ export class DataAggregatingRouteHandler extends RouteHandler {
       (obj, countryCode, i) => ({ ...obj, [countryCode]: countryAccessList[i] }),
       {},
     );
-    dataSourceEntities = dataSourceEntities.filter(e => countryAccess[e.country_code]);
+    const permittedDataSourceEntities = dataSourceEntities.filter(
+      e => countryAccess[e.country_code],
+    );
+
+    return dataSourceEntityFilter
+      ? filterEntities(permittedDataSourceEntities, dataSourceEntityFilter)
+      : permittedDataSourceEntities;
+  };
+
+  getBaseDataSourceEntities = async (entity, entityType, options) => {
+    const { includeSiblingData, aggregationEntityType } = options;
+    const hierarchyId = await this.fetchHierarchyId();
+    if (!entityType) return entity.getNearestOrgUnitDescendants(hierarchyId);
+
+    /**   A
+     *   / \
+     *  B   C
+     * This is when we are at B and want to fetch data for both B and C
+     */
+    if (includeSiblingData) {
+      const ancestor = await entity.getAncestorOfType(aggregationEntityType, hierarchyId);
+      return ancestor.getDescendantsOfType(entityType, hierarchyId);
+    }
```

Comment:
This should be the only functional change in this file

Reply:
I see you need to make a functional change in here - hoping it's easy enough to do that in the original form of `fetchDataSourceEntities`